### ISSUE TITLE
[polymetis] mass matrix

### ISSUE THREAD

**Issue by minoring** [facebookresearch/fairo#1059]
## Type of Issue

Select the type of issue:
- [ ] Bug report (to report a bug)
- [x] Feature request (to request an additional feature)
- [ ] Tracker (I am just using this as a tracker)
- [ ] Refactor request
- [ ] Documentation Ask

## Description

I would like to do operational space control (OSC) using polymetis in a real franka panda robot, and the mass matrix is necessary to compute this. In my knowledge, there is no feature to get a mass matrix in polymetis yet, and asking it as a new feature. (Or please let me know if we can get mass matrix now.)

## Expected Behavior

We can get mass matrix from libfranka, like joint_positions or joint_velocities.
https://github.com/facebookresearch/fairo/blob/main/polymetis/polymetis/protos/polymetis.proto#L105

**Comment by exhaustin** [comment]
Hello @minoring! The mass matrix is not part of the raw [franka::RobotState](https://frankaemika.github.io/libfranka/structfranka_1_1RobotState.html) provided by libfranka, so we do not intend to provide that within the raw robot states since that would incur additional computation load when running any controller.

The good news is that we have a [robot model within TorchControl](https://facebookresearch.github.io/fairo/polymetis/torchcontrol-doc.html#models),  which can be used within custom controllers. TorchControl also comes with a [Cartesian impedance control](https://github.com/facebookresearch/fairo/blob/0b07562e2498c55c689a7345d45f474a42464e14/polymetis/polymetis/python/torchcontrol/policies/impedance.py#L76) implementation, which should be a great example of how the robot model can be used for operational space control (even though the mass matrix is not exposed explicitly, inverse dynamics is sufficient in this use case).

If this method of gaining access to robot dynamics within the controller looks good to you, but inverse dynamics isn't enough for your use case and you still need the full mass matrix regardless, I am happy to see if we can expose that from the robot model.

**Comment by minoring** [comment]
Thanks @exhaustin for giving a reply for this. And thanks for citing the parts  that are related, and explaining why the mass matrix is not part of Polymetis. I think it make sense. For my case, a full mass matrix is required, so I computed it on the server with some overhead in my version of Polymetis. But I also think that this could not be the feature everyone needs, so I did not make any pull request for that. As you mentioned, the way to get a mass matrix without overhead is getting it from the robot model. I hope there will be features in the robot model  too :)
